javascript/new-src/src/factories/ola.js

```diff
@@ -22,6 +22,29 @@
 ola.factory('$ola', ['$http', '$window', 'OLA',
   function($http, $window, OLA) {
     'use strict';
+    // holds the highest channel that was used by faders or the keypad
+    var highestChannelNumberUsed = 0;
+
+    // Search for the highest channel in the array `dmx`
+    // that having a value greater than MIN_CHANNEL_VALUE
+    // and update `highestChannelNumberUsed` if needed.
+    //
+    // Only channels higher than the current `highestChannelNumberUsed`
+    // will be checked.
+    var updateHighestChannelNumberUsed = function(dmx) {
+      for (var channel = dmx.length; channel > highestChannelNumberUsed;
+           channel--) {
+
+        var value = parseInt(dmx[channel - 1], 10);
```

Comment:
This should probably call `channelValueCheck` to sanity check the value the field was set to. But then `updateHighestChannelNumberUsed` is just `dmxConvert` without the return, so perhaps we should just have one function?